include/common_structs.h

```diff
@@ -5,62 +5,92 @@
 #include "types.h"
 
 typedef struct {
-    /* 0x000 */ s16 unk_00;
-    /* 0x002 */ s16 unk_02;
-    /* 0x004 */ char unk_04[0x44];
+    /* 0x000 */ u32 currentButtons;
+    /* 0x004 */ u32 altCurrentButtons; /* used for battle when flag 80000 set */
+    /* 0x008 */ char unk_08[8];
+    /* 0x010 */ u32 pressedButtons; /* 1 for the frame button begins to be held */
+    /* 0x014 */ u32 altPressedButtons; /* used for battle when flag 80000 set */
+    /* 0x018 */ char unk_18[8];
+    /* 0x020 */ u32 heldButtons; /* 1 every 4th frame during hold */
+    /* 0x024 */ u32 altHeldButtons; /* used for battle when flag 80000 set */
+    /* 0x028 */ char unk_28[8];
+    /* 0x030 */ u32 prevButtons; /* from previous frame */
+    /* 0x034 */ char unk_34[12];
+    /* 0x040 */ u8 stickX; /* with deadzone */
+    /* 0x041 */ u8 altStickX; /* used for battle when flag 80000 set */
+    /* 0x042 */ char unk_42[2];
+    /* 0x044 */ u8 stickY; /* with deadzone */
+    /* 0x045 */ u8 altStickY; /* used for battle when flag 80000 set */
+    /* 0x046 */ char unk_46[2];
     /* 0x048 */ u16 unk_48[4];
     /* 0x050 */ u16 unk_50[4];
     /* 0x058 */ char unk_58[0x10];
     /* 0x068 */ u16 demoButtonInput;
     /* 0x06A */ s8 demoStickX;
     /* 0x06B */ s8 demoStickY;
-    /* 0x06C */ char unk_6C[0x4];
-    /* 0x070 */ s8 unk_70;
-    /* 0x071 */ s8 demoState;
-    /* 0x072 */ s8 unk_72;
+    /* 0x06C */ u32* mainScriptID;
+    /* 0x070 */ u8 isBattle;
```

Comment:
most likely the source of the nonmatching is something like one of these which you've changed from signed to unsigned. Be careful with this, as this can be the difference between, for example, ``lb`` and ``lbu``

Reply:
Yep, it was a combination of signedness differences are misalignment. It matches now :tada: